src/lib.rs

```diff
@@ -486,86 +481,93 @@ pub(crate) fn add_missing_provided_values(
         liquid_object.insert(k.clone().into(), value);
         Ok(())
     })?;
-    Ok(liquid_object)
+    Ok(())
 }
 
-fn merge_conditionals(
-    template_config: &Config,
-    liquid_object: liquid::Object,
+fn fill_placeholders_and_merge_conditionals(
+    config: &mut Config,
+    mut liquid_object: liquid::Object,
```

Comment:
`liquid_object` is now mutable, so why are we still need to return a `liquid::Object` then?

With a mutable input we could set and fill all the things we need, without returning a new `Object`

Reply:
because we need to take ownership!

The `Rhai` API needs a `Rc<Cell<Object>> + 'static` - thus we need to pass ownership whenever we need the engine to resove stuff. It's either that (like we did before my changes here), convert the var into Rc<T> like I did originally in this PR, or resorting to `unsafe` which I wont do.

Or you can try your hand at introducing the proper lifetime specs in `Rhai` - I gave up the other day - tried but gave up as it would introduce major API breakning which doubt they want.

Reply:
@sassman ?

Reply:
I See, yeah lifetime hell is also unpleasant.